### DISCUSSION TITLE
Double click signal for Button?

### DISCUSSION THREAD

**Discussion by feois** [godotengine/godot-proposals#8730] [GUI]
Make a signal like `Button.pressed()` but for double click. I know it can be done through `_input()` and [double_click](https://docs.godotengine.org/en/stable/classes/class_inputeventmousebutton.html#class-inputeventmousebutton-property-double-click) but a signal can be a better enhancement

**Comment by Mickeon** [comment]
What's the use-case of a button that needs to be double-clicked? How common is it?

**Reply by feois** [reply]
Basically if single click then effect A, if double click then depending on the situation, either just effect B or effect A + B. If you want use case, something like a skill button that when clicked trigger skill effect but also trigger extra effects when double clicked under certain circumstances? I guess it might not be too common.

**Reply by romlok** [reply]
From my experience, "click to select" combined with "double-click to select and confirm" is a fairly common pattern in video game UIs?

Such as a level-select UI, where clicking a level name/thumbnail selects it and also shows more information about that level, while double-clicking would start the level immediately, the same way as clicking the "start game" button.

**Comment by AThousandShips** [comment]
How would this handle the new signal? If the `pressed` signal is *not* called for double clicks by default this breaks compatibility

**Reply by feois** [reply]
A new property that determine pressed is emitted or not with default being yes since backward compatibility

**Comment by theraot** [comment]
For people finding this: you can get `InputEventMouseButton` in `_gui_input` or the `gui_event` signal. That way you know the the interaction is on your `Control` (unlike with `_input`). And, of course, you can read the [`double_click`](https://docs.godotengine.org/en/stable/classes/class_inputeventmousebutton.html#class-inputeventmousebutton-property-double-click) property. Bonus: Since it is not a `Button` thing, you can implement double click on other controls (e.g. a `Label` or a `TextEdit`).
